returnn/torch/distributed.py

```diff
@@ -39,10 +39,10 @@ def __init__(self, options: Dict[str, Any]):
         else:
             import torch.distributed as dist
 
-            # when no backend is specified, both gloo and nccl backends will be created
-            # the gloo backend will be used for collectives with CPU tensors and
-            # the nccl backend will be used for collectives with CUDA tensors
-            dist.init_process_group(backend=self._opts.get("backend", None))
+            # When no backend is specified, we set gloo for CPU tensors and nccl for CUDA tensors as backend.
+            # torch 2.6.0 and onwards require explicitly setting the backends.
+            # See https://github.com/rwth-i6/returnn/issues/1724 for discussion.
+            dist.init_process_group(backend=self._opts.get("backend", "cpu:gloo,cuda:nccl"))
```

Comment:
The backend needs to be retrieved outside of the else branch otherwise if this branch is taken and the backend is set explicitly the check_no_unknown_opts() function below will throw an assertion error.
This leads me to believe nobody ever set the backend explicitly as it's currently broken.